src/server.ts

```diff
@@ -68,59 +97,90 @@ export const appRouter = trpc.router({
             .optional()
             .default(''),
         }),
-      ),
-      (params) => {
+      )
+      .resolve((params) => {
         return {
           greeting: 'hello ' + params.ctx.user?.id ?? params.input.hello,
         };
-      },
-    ),
+      }),
     // procedure with auth
-    viewerWhoami: trpc.resolver(
+    viewerWhoAmi: procedure.use(isAuthed).resolve(({ ctx }) => {
       // `isAuthed()` will propagate new `ctx`
-      isAuthed(),
-      ({ ctx }) => {
-        // `ctx.user` is now `NonNullable`
-        return `your id is ${ctx.user.id}`;
-      },
-    ),
-    // a bit iffy - if you want to use it without `trpc.resolver()`
-    aQueryWithoutResolverWrapper: () => {
-      return {
-        ok: true,
-        data: 'test',
-      };
-    },
+      // `ctx.user` is now `NonNullable`
+      return `your id is ${ctx.user.id}`;
+    }),
   },
+
   mutations: {
     // mutation with auth + input
-    postAdd: trpc.resolver(
-      trpc.zod(
+    postAdd: procedure
+      .input(
         z.object({
           title: z.string(),
           body: z.string(),
         }),
-      ),
-      isAuthed(),
-      ({ input, ctx }) => {
+      )
+      .use(isAuthed)
+      .resolve(({ ctx, input }) => {
         const post: typeof postsDb[number] = {
           ...input,
           id: `${Math.random()}`,
           userId: ctx.user.id,
         };
         postsDb.push(post);
+        return post;
+      }),
+    fireAndForget: procedure.input(z.string()).resolve(() => {
+      // no return
+    }),
+    editOrg: procedure
+      .apply(
+        isPartofOrg(
+          z.object({
+            organizationId: z.string(),
+            data: z
+              .object({
+                name: z.string(),
+              })
+              .partial(),
+          }),
+        ),
```

Comment:
I don't love the name `.apply()` here - any ideas for something better?

Reply:
Feels like a sort of plugin to me. Maybe  `.register` ([like Fastify](https://www.fastify.io/docs/latest/Reference/Plugins/))?

Reply:
Could `.from(…)` make sense? Like you’re building a procedure from another [partial] procedure [builder]?

Reply:
Or maybe `.assign()` or `.extend()` as it works similarly to `Object.assign()`

Reply:
I made it part of the `.use()`-fn! How does that feel to you?

Reply:
Or maybe `concat` would be a good word too, like `[].concat`. Probably less confusing if I have it separate to `use`

Reply:
Anywayyyy I shouldn't get stuck on naming, can be changed easily with deprecation

Reply:
Merging easier than splitting though! I’d be basically fine with any of the above but agree that overloading use could be confusing.